File: TP01/TP01Q10 - Arquivo em C/Arquivo.c
//TP01Q10 - Arquivo em java
//@Author: Vitoria Simil Araujo

#include "stdio.h"
#include "stdlib.h"
#include "string.h"

//fuctions
void execute(int size);

int main(void) {
    //data
    FILE *file = fopen("file.txt", "wb"); //using binary file
    char input[1000] = {};
    float number = 0.0;
    
    //read the input archive
    scanf("%[^\n]", input);
    getchar();

    int n = atoi(input); //number of numbers to be read
    int size = n;

    while ( n > 0 ) {
        scanf("%[^\n]", input);
        getchar();
        number = atof(input);
        fwrite(&number, sizeof(float), 1, file);
        n--;
    }
    
    //close file
    fclose(file);

    //execute
    execute(size);
}

//method to execute
void execute(int size) {
    //data
    FILE *file = fopen("file.txt", "rb");
    float number = 0.0;
    
    for ( int i = size; i > 0; i-- ) {
        //pointing seek
        fseek(file, sizeof(float) * ( i - 1 ), SEEK_SET);
        fread(&number, sizeof(float), 1, file);
        printf("%g\n", number);
    }

    //close file
    fclose(file);
}

File: TP01/TP01Q09 - Arquivo em Java/Arquivo.java
//TP01Q09 - Arquivo em java
//@Author: Vitoria Simil Araujo

import java.io.RandomAccessFile;

class Arquivo {
    //method to execute
    public static void execute(int size) throws Exception {
        //data
        RandomAccessFile readFile = new RandomAccessFile("file.txt", "r");
        double number = 0.0;

        for(int i = size; i > 0; i--){
			    //pointing seek
            readFile.seek((9 * (i - 1)));
			      number = readFile.readDouble();

            if ( number - (int)number == 0 ) {
                MyIO.println((int)number);
            }
            else {
                MyIO.println(number);
            }
            
			  readFile.readByte();
		}

        //close file
		    readFile.close();
    }

    public static void main(String[] args) throws Exception {
        //data
        RandomAccessFile readFile = new RandomAccessFile("file.txt", "rw");
        int n = 0, size = 0;
        double number;

        //read the input archive
        n = MyIO.readInt(); //number of numbers to be read
        size = n;

        while ( n > 0 )
        {
            number = MyIO.readDouble();
            readFile.writeDouble(number);
            readFile.writeBytes("\n");
            n--;
            
        }
        
        //close file
        readFile.close();
        execute(size);
    }
}
